### PR TITLE
android task 3  add event logging to instrument the app

### PR DESCRIPTION
**Pull request by lttsNilesh**
Add event logging to instrument the app - log an event for todo_created, todo_deleted, and list_viewed. These should be the same names across all apps. Add an “app opened” and “app backgrounded” log event. For Android/iOS only.

### CODE REVIEW

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt [hunk 1]
```diff
@@ -13,6 +13,13 @@ object StatsIgUtil {
 
     private const val FEATURE_GATE_DELETE_TODO_ACCESS = "enable_delete_todo"
     private const val TEST_MOBILE_DYNAMIC_CONFIG = "mobile_dynamic_config"
+    const val LOG_EVENT_APP_OPENED = "app opened"
+    const val LOG_EVENT_APP_BACKGROUNDED = "app backgrounded"
```
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:17]: be consistent here - use_underscores inbetween words for event names

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt [hunk 2]
```diff
@@ -43,4 +50,26 @@ object StatsIgUtil {
...
+    fun eventLogWithoutMetadata(eventName: String) {
+        Statsig.logEvent(eventName)
+    }
+
+    /**
+     * The function to register the event with metadata in the Statsig portal.
+     *
+     * @param eventName Name of the event to be logged.
+     */
+    fun eventLogWithMetadata(
```
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:67]: these two functions dont add anything - just use `Statsig.logEvent` directly in the callsite

Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:67]: in general, the parameters to your function are documentation enough, adding a `withMetadata` is unnecessary.

But again, just delete these functions

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt [hunk 3]
```diff
@@ -13,6 +13,13 @@ object StatsIgUtil {
 
     private const val FEATURE_GATE_DELETE_TODO_ACCESS = "enable_delete_todo"
     private const val TEST_MOBILE_DYNAMIC_CONFIG = "mobile_dynamic_config"
+    const val LOG_EVENT_APP_OPENED = "app opened"
```
Comment (tore-statsig) [inline android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt:16]: I'd name these `APP_OPENED`, `TODO_CREATED`, etc.  The `LOG_EVENT` piece is redundant

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 1 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: Sure, will do it.

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 2 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: I will use Statsig.logEvent directly.

Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: Will remove it.

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 3 above)
Comment (lttsNilesh) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: I will make the change

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 1 above)
Comment (tore-statsig) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: Done in https://github.com/statsig-io/samples/pull/5 it seems

File: android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt (hunk 2 above)
Comment (tore-statsig) [reply android-todo-list/architecture-samples-main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/StatsIgUtil.kt]: Done in https://github.com/statsig-io/samples/pull/5

**Comment by tore-statsig** [comment]
Can close this, since all the work is in https://github.com/statsig-io/samples/pull/5 ?

**Pull request merged**